### PR TITLE
feat: add matrix strategy

### PR DESCRIPTION
**Pull request by abhijeetSaroha**
Add support of matrix.

Solves #81

### CODE REVIEW

**Comment by abhijeetSaroha** [comment]
While the combination logic successfully creates all 6 combinations, the output seems to be referencing the matrix variables without properly substituting their values. The output looks like this for each combination:

```
[Setup] Environment: {{ matrix.env }}, Architecture: {{ matrix.arch }}
[Setup] Using base image: {{ vars.base_image }}
[Setup] Installing dependencies for {{ matrix.env }}
```

Tried using `${{ matrix.env }}` or `${{ matrix.arch }}` for variable substitution, but this resulted in a "matrix variable not found" error.

Additionally, commands like `makim tests.smoke` that were previously functional are no longer running after this change.

**Comment by abhijeetSaroha** [comment]
It works fine now as it should be.

I also break the function `_run_command` in two functions as you said. I have create another function for **MakimLogs execution**.

https://github.com/abhijeetSaroha/makim/blob/matrix-strategy/src/makim/core.py#L455C2-L480C42

File: src/makim/core.py [hunk 1]
```diff
@@ -518,20 +579,22 @@ def _run_hooks(self, args: dict[str, Any], hook_type: str) -> None:
...
                 '`vars` attribute inside the group '
                 f'{self.group_name} is not a dictionary.',
                 MakimError.MAKIM_VARS_ATTRIBUTE_INVALID,
             )
 
-        env, variables = self._load_scoped_data('task')
-        for k, v in env.items():
-            os.environ[k] = v
+        # Get matrix configuration if it exists
+        matrix_combinations: list[dict[str, Any]] = [{}]
```
Comment (xmnlab) [inline src/makim/core.py:591]:
```suggestion
        matrix_combinations: list[dict[str, Any]] = []
```

File: src/makim/core.py [hunk 2]
```diff
@@ -451,6 +452,66 @@ def _load_task_args(self) -> None:
...
+        matrix_config : dict
+            Dictionary containing matrix variables and their possible values
+
+        Returns
+        -------
+        list[dict]
+            List of dictionaries, each containing one possible combination
+        """
+        if not matrix_config:
+            return [{}]
```
Comment (xmnlab) [inline src/makim/core.py:498]:
```suggestion
            return []
```

File: src/makim/core.py [hunk 3]
```diff
@@ -565,36 +628,47 @@ def _run_command(self, args: dict[str, Any]) -> None:
...
-            args=args_input, env=env, vars=variables
-        )
         width, _ = get_terminal_size()
 
-        if self.verbose:
-            MakimLogs.print_info('=' * width)
-            MakimLogs.print_info(
-                'TARGET: ' + f'{self.group_name}.{self.task_name}'
+        # Run command for each matrix combination
+        for matrix_vars in matrix_combinations:
```
Comment (xmnlab) [inline src/makim/core.py:634]:
```suggestion
        for matrix_vars in matrix_combinations or [{}]:
```

**Comment by abhijeetSaroha** [comment]
@xmnlab, I have made the changes and rebase the branch too.

**Comment by xmnlab** [comment]
thank you @abhijeetSaroha ! great job!

**Pull request merged**